### ISSUE TITLE
Falling state when standing on the edge of tile.

### ISSUE THREAD

**Issue by RippeR37** [michlord/team-project-2014#47]
You cant do slash or jump in that state, just move right/left. Needs to be fixed.

Example:

![Bug](http://i.imgur.com/u7wMqBp.png)

**Comment by RippeR37** [comment]
As animations have varying size (width is important here), if you **stand** near ledge and try to slash, animation is cut before it finishes due to resizing of animation and this bug (chaning state to falling for a moment and then back to standing).

**Comment by michlord** [comment]
This requires quite a bit of work... if you want to do it correctly.


One way to fix this is to have constant-size animations.

Other fix would be to introduce some epsilon to collision checking like this:
```
void OnGround::onUpdate(CharacterEntity *entity){
        sf::Vector2f feet = entity->getFeetPosition();

        if(false == entity->level->isFeetOnGround(feet) &&
           false == entity->level->isFeetOnGround(feet + sf::Vector2f(5.0f, 0.0f)) &&
           false == entity->level->isFeetOnGround(feet - sf::Vector2f(5.0f, 0.0f)) ) {
            entity->movementSM->changeState(new Fall());
            entity->globalMovementSM->changeState(new InAir());
        }
}
```
This gets rid of the endless loop between falling and onground state changes.
It doesn't allow slashing though (as soon as it starts the feet moves too far and it starts falling - one way to fix is to increase epsilon - and this is what I will do). 

The correct way would be to rework states a little bit but I don't have enough time to do this.

**Comment by PiotrPytlik** [comment]
I think it's still a MUST to change the sizes of the player and knight animations, because the different animation sizes cause too many problems, such as odd stretching and shrinking of the player.

**Comment by RippeR37** [comment]
I believe it's fixed with @michlord commit a717fa3.
